ivy/utils/backend/sub_backend_handler.py

```diff
@@ -0,0 +1,148 @@
+import os
+import importlib
+from types import ModuleType, FunctionType
+
+import ivy
+from ivy.func_wrapper import _wrap_function
+from ivy.utils.exceptions import IvyException
+
+_sub_backend_dict = dict()
+_backend_to_sub_backends_dict = dict()
+# torch sub_backends
+
+_sub_backend_dict["xformers"] = "ivy.functional.backends.torch.sub_backends.xformers"
+
+
+_backend_to_sub_backends_dict["torch"] = ["xformers"]
+
+
+_all_sub_backends = _backend_to_sub_backends_dict["torch"]
+
+
+original_backend_dict = None
+
+
+def set_sub_backend(sub_backend_str: str):
+    if ivy.backend == "none":
+        print("You must set a backend first:")
+        available_sub_backends()
```

Comment:
let's use `warning.logging` module here, also we shouldn't be printing all the subbackends for all available backends. the list of available subbackends should be shown only when a backend is set.